search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/prompt/PromptUtil.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative.prompt;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.commons.text.StringEscapeUtils;
+
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * TODO Should prompt engineering llm-specific?
+ *
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class PromptUtil {
```

Comment:
A high level question, should prompt functionality be a generic component in ml-commons, not only for conversational search pipelines?

Reply:
If there is a plan for prompt template management (for the 2.11 release), I see this moving out of search pipeline and into a common spot within ml-commons.

Reply:
+1 we don't have to design it here but feels like it should be generic utility exposed by ml-commons. 
Lets open an issue to discuss?

Reply:
#1161 doesn't address it?

Reply:
It does, thank you!